consensus/enclave/impl/build.rs

```diff
@@ -25,4 +26,41 @@ fn main() {
 
     fs::write(&env.out_dir().join("target_features.rs"), &contents)
         .expect("Could not write target feature array");
+
+    rerun_if_env_changed!("FEE_SPEND_PUBLIC_KEY");
+    rerun_if_env_changed!("FEE_VIEW_PUBLIC_KEY");
+
+    let mut fee_spend_public_key = [0u8; 32];
+    let mut fee_view_public_key = [0u8; 32];
+
+    let default_fee_spend_pub = "26b507c63124a2f5e940b4fb89e4b2bb0a2078ed0c8e551ad59268b9646ec241";
+    let default_fee_view_pub = "5222a1e9ae32d21c23114a5ce6bb39e0cb56aea350d4619d43b1207061b10346";
+
+    // Check for env var and override
+    fee_spend_public_key[..32].copy_from_slice(
+        &hex::decode(&var("FEE_SPEND_PUBLIC_KEY").unwrap_or(default_fee_spend_pub.to_string()))
+            .unwrap(),
+    );
+    fee_view_public_key[..32].copy_from_slice(
+        &hex::decode(&var("FEE_VIEW_PUBLIC_KEY").unwrap_or(default_fee_view_pub.to_string()))
+            .unwrap(),
+    );
+
+    let mut constants = format!(
+        "pub const FEE_SPEND_PUBLIC_KEY: [u8; 32] = {:?};",
+        fee_spend_public_key
+    );
+    constants.push_str(&format!(
+        "pub const FEE_VIEW_PUBLIC_KEY: [u8; 32] = {:?};",
+        fee_view_public_key
+    ));
+
+    let mut search_path =
+        PathBuf::try_from(var("CARGO_MANIFEST_DIR").expect("Could not read CARGO_MANIFEST_DIR"))
+            .expect("Could not construct PathBuf from CARGO_MANIEFST_DIR")
+            .canonicalize()
+            .expect("Could not canonicalize CARGO_MANIFEST_DIR");
+    search_path.push("src");
```

Comment:
We shouldn't be generating files under `src`. It should go inside `target/`. There's an example of how this is done in https://github.com/mobilecoinofficial/mobilecoin/blob/master/util/build/grpc/src/lib.rs#L15

Reply:
I think it has to be under src so that it can get compiled into the code. It's using `mod constants` for example, in `lib.rs`. It's still in the build environment, so it doesn't write the constants.rs out to the source repo.

Reply:
This is how you can include generated source code in the target dir: https://github.com/mobilecoinofficial/mobilecoin/blob/47679adf3b45674655c51a88666851a1649266cc/util/grpc/src/lib.rs#L12

Reply:
Fixed up - I misunderstood out the out_dir worked!